### PR TITLE
feat(language-service): add style scoped and module completion

### PR DESCRIPTION
**Pull request by runyasak**
# Updated
* add `style module` completion with a new screenshot
* ~~change to `style module` completion instead of `style scoped` with a new screenshot~~

# Overview

I've added the `scoped` and `module` attributes to style completion items.
Now, when we type `style` in completion, `scoped` and `module` are available as options.

# Screenshots

https://github.com/user-attachments/assets/208847d7-e9f9-4c82-a893-91a52ee4caf2

---

This is my first PR. If you have any feedback or suggestions, please do not hesitate to let me know.

### CODE REVIEW

**Comment by KazariEX** [comment]
I think there is no need to extract languages to the top level, as it might be obtained from providers in the future. #4645 

What about `module`?

**Comment by runyasak** [comment]
> I think there is no need to extract languages to the top level, as it might be obtained from provider in the future. #4645
> 
> What about `module`?

Hi @KazariEX, I apologize for not seeing your PR earlier.
Are you referring to [CSS Modules](https://vuejs.org/api/sfc-css-features.html#css-modules)?
If no one is assigned to this feature, may I contribute to it?
Should I create a new PR or update this one?

**Comment by KazariEX** [comment]
Just update directly, I think they belong to the same feature.

**Comment by runyasak** [comment]
@KazariEX I've updated my PR to `module`.
Please review my code again.

**Comment by KazariEX** [comment]
Sorry I didn't mean replacement but both `scoped` and `module`.

File: packages/language-service/lib/plugins/vue-sfc.ts [hunk 1]
```diff
@@ -232,3 +226,20 @@ export function create(): LanguageServicePlugin {
 		}
 	}
 }
+
+function getStyleCompletionItem(
+	styleItem: vscode.CompletionItem,
+	lang: string,
+	module = false,
```
Comment (KazariEX) [inline packages/language-service/lib/plugins/vue-sfc.ts:233]:
```suggestion
	attr = '',
```

File: packages/language-service/lib/plugins/vue-sfc.ts [hunk 2]
```diff
@@ -232,3 +226,20 @@ export function create(): LanguageServicePlugin {
...
+): vscode.CompletionItem {
+	return {
+		...styleItem,
+		kind: 17 satisfies typeof vscode.CompletionItemKind.File,
+		detail: lang === 'postcss' ? '.css' : `.${lang}`,
+		label: styleItem.label + ' lang="' + lang + '"' + (module ? ' module' : ''),
```
Comment (KazariEX) [inline packages/language-service/lib/plugins/vue-sfc.ts:239]:
```suggestion
		label: styleItem.label + ' lang="' + lang + '"' + (attr ? ' ' + attr : ''),
```

File: packages/language-service/lib/plugins/vue-sfc.ts [hunk 3]
```diff
@@ -232,3 +226,20 @@ export function create(): LanguageServicePlugin {
...
+		textEdit: styleItem.textEdit ? {
+			...styleItem.textEdit,
+			newText: styleItem.textEdit.newText + ' lang="' + lang + '"' + (module ? ' module' : ''),
```
Comment (KazariEX) [inline packages/language-service/lib/plugins/vue-sfc.ts:242]:
```suggestion
			newText: styleItem.textEdit.newText + ' lang="' + lang + '"' + (attr ? ' ' + attr : ''),
```

**Comment by runyasak** [comment]
> Sorry I didn't mean replacement but both `scoped` and `module`.

No problem. I will update now.

**Review by KazariEX — CHANGES_REQUESTED** [review]
Maybe we can do this.

**Comment by runyasak** [comment]
@KazariEX  I've updated my PR with `attr` as your suggestion.
Please review my PR again.

**Review by KazariEX — APPROVED** [review]

**Comment by johnsoncodehk** [comment]
LGTM, thanks!

**Pull request merged**